packages/client/ui/react-ui/src/providers/CrossmintWalletProvider.tsx

```diff
@@ -35,17 +50,22 @@ export type WalletConfig = WalletParams & { type: "evm-smart-wallet" };
 export function CrossmintWalletProvider({
     children,
     defaultChain,
+    enablePasskeyPrompt = false,
 }: {
     children: ReactNode;
     defaultChain: EVMSmartWalletChain;
+    enablePasskeyPrompt?: boolean;
 }) {
     const { crossmint } = useCrossmint("CrossmintWalletProvider must be used within CrossmintProvider");
     const smartWalletSDK = useMemo(() => SmartWalletSDK.init({ clientApiKey: crossmint.apiKey }), [crossmint.apiKey]);
 
-    const [state, setState] = useState<ValidWalletState>({ status: "not-loaded" });
+    const [walletState, setWalletState] = useState<ValidWalletState>({ status: "not-loaded" });
+    const [passkeyPromptState, setPasskeyPromptState] = useState<PasskeyPromptState>({ open: false });
 
-    const getOrCreateWallet = (config: WalletConfig = { type: "evm-smart-wallet", signer: { type: "PASSKEY" } }) => {
-        if (state.status == "in-progress") {
+    const getOrCreateWallet = async (
```

Comment:
the amount of nesting, and closure stuff we need to keep track of in here is a bit of a smell to me, maybe worth re-evaluating how this is setup.

I'm off on PTO but @AlbertoElias would like you're take here

Reply:
I've updated this to be more imperative and less nested

Reply:
Much better!

Reply:
Thanks! @AlbertoElias